contracts/EVMScriptFactories/AlterTierInOperatorGrid.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-FileCopyrightText: 2025 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.8.6;
+
+import "../TrustedCaller.sol";
+import "../libraries/EVMScriptCreator.sol";
+import "../interfaces/IEVMScriptFactory.sol";
+import "../interfaces/IOperatorGrid.sol";
+
+/// @author dry914
+/// @notice Creates EVMScript to alter a tier in OperatorGrid
+contract AlterTierInOperatorGrid is TrustedCaller, IEVMScriptFactory {
+
+    // -------------
+    // ERRORS
+    // -------------
+
+    error TierNotExists();
+    error WrongCalldataLength(uint256 length);
+
+    // -------------
+    // VARIABLES
+    // -------------
+
+    /// @notice Address of OperatorGrid
+    IOperatorGrid public immutable operatorGrid;
+
+    // -------------
+    // CONSTRUCTOR
+    // -------------
+
+    constructor(address _trustedCaller, address _operatorGrid)
+        TrustedCaller(_trustedCaller)
+    {
+        operatorGrid = IOperatorGrid(_operatorGrid);
+    }
+
+    // -------------
+    // EXTERNAL METHODS
+    // -------------
+
+    /// @notice Creates EVMScript to alter a tier in OperatorGrid
+    /// @param _creator Address who creates EVMScript
+    /// @param _evmScriptCallData Encoded: uint256 _tierId, IOperatorGrid.TierParams _tierParams
+    function createEVMScript(address _creator, bytes memory _evmScriptCallData)
+        external
+        view
+        override
+        onlyTrustedCaller(_creator)
+        returns (bytes memory)
+    {
+        (uint256 _tierId, IOperatorGrid.TierParams memory _tierParams) = _decodeEVMScriptCallData(_evmScriptCallData);
+
+        _validateInputData(_tierId);
+
+        return
+            EVMScriptCreator.createEVMScript(
+                address(operatorGrid),
+                IOperatorGrid.alterTier.selector,
+                _evmScriptCallData
+            );
+    }
+
+    /// @notice Decodes call data used by createEVMScript method
+    /// @param _evmScriptCallData Encoded: uint256 _tierId, IOperatorGrid.TierParams _tierParams
+    /// @return Tier ID and tier parameters which should be updated in operator grid
+    function decodeEVMScriptCallData(bytes memory _evmScriptCallData)
+        external
+        pure
+        returns (uint256, IOperatorGrid.TierParams memory)
+    {
+        return _decodeEVMScriptCallData(_evmScriptCallData);
+    }
+
+    // ------------------
+    // PRIVATE METHODS
+    // ------------------
+
+    function _decodeEVMScriptCallData(bytes memory _evmScriptCallData)
+        private
+        pure
+        returns (uint256, IOperatorGrid.TierParams memory)
+    {
+        if (_evmScriptCallData.length != 160) revert WrongCalldataLength(_evmScriptCallData.length);
+        return abi.decode(_evmScriptCallData, (uint256, IOperatorGrid.TierParams));
+    }
+
+    function _validateInputData(uint256 _tierId) private view {
+        IOperatorGrid.Tier memory tier = operatorGrid.tier(_tierId);
+        if (tier.operator == address(0)) revert TierNotExists();
```

Comment:
is redundant since operatorGrid.tier(_tierId) already reverts internally if the tier does not exist.

and now there are now way to check exists tier properly - needs to add `tiersCount` in OperatorGrid